vignettes/irecist.Rmd

```diff
@@ -0,0 +1,914 @@
+        count_vals(var = AVALC.join, val = "iCR") == 0 |
+        count_vals(var = AVALC.join, val = "iPR") == 0
+    ),
+  set_values_to = exprs(AVALC = "iPR")
+)
+```
+
+Please note that the result `AVALC = "Y"` for confirmed clinical benefit is
+defined by the first _two_ events specified for `events`. For subjects with
+observations fulfilling both events the one with the earlier date should be
+selected (and not the first one in the list).
+
+```{r}
+adrs <- adrs %>%
+  derive_extreme_event(
+    by_vars = exprs(STUDYID, USUBJID),
+    order = exprs(desc(AVALC), ADT),
+    mode = "first",
+    source_datasets = list(
+      ovr = ovr,
+      adsl = adsl
+    ),
+    events = list(icrsp_y_icr, icrsp_y_ipr, no_data_n),
+    set_values_to = exprs(
+      PARAMCD = "ICRSP",
+      PARAM = "iRECIST Confirmed Response by Investigator",
+      PARCAT1 = "Tumor Response",
+      PARCAT2 = "Investigator",
+      PARCAT3 = "iRECIST",
+      AVAL = yn_to_numeric(AVALC),
+      ANL01FL = "Y"
+    )
+  )
+
+adrs <- adrs %>%
+  derive_extreme_event(
+    by_vars = exprs(STUDYID, USUBJID),
+    order = exprs(desc(AVALC), ADT),
+    mode = "first",
+    events = list(icrsp_y_icr, icrsp_y_ipr, icb_y, no_data_n),
+    source_datasets = list(
+      ovr = ovr,
+      adsl = adsl
+    ),
+    set_values_to = exprs(
+      PARAMCD = "ICCB",
+      PARAM = "iRECIST Confirmed Clinical Benefit by Investigator",
+      PARCAT1 = "Tumor Response",
+      PARCAT2 = "Investigator",
+      PARCAT3 = "iRECIST",
+      AVAL = yn_to_numeric(AVALC),
+      ANL01FL = "Y"
+    ),
+    check_type = "none"
+  )
+
+adrs <- adrs %>%
+  derive_extreme_event(
+    by_vars = exprs(STUDYID, USUBJID),
+    tmp_event_nr_var = event_nr,
+    order = exprs(event_nr, ADT),
+    mode = "first",
+    events = list(icbor_icr, icbor_ipr, ibor_isd, ibor_non_icriupd, ibor_icpd, ibor_iupd, ibor_ne, no_data_missing),
+    source_datasets = list(
+      ovr = ovr,
+      adsl = adsl
+    ),
+    set_values_to = exprs(
+      PARAMCD = "ICBOR",
+      PARAM = "iRECIST Best Confirmed Overall Response by Investigator",
+      PARCAT1 = "Tumor Response",
+      PARCAT2 = "Investigator",
+      PARCAT3 = "iRECIST",
+      AVAL = aval_resp(AVALC),
+      ANL01FL = "Y"
+    )
+  )
+```
+
+```{r, eval=TRUE, echo=FALSE}
+dataset_vignette(
+  adrs,
+  display_vars = exprs(USUBJID, AVISIT, PARAMCD, PARAM, AVALC, ADT, RANDDT, ANL01FL),
+  filter = PARAMCD %in% c("ICRSP", "ICCB", "ICBOR")
+)
+```
+## Other Endpoints {#other}
+
+The following parameters may also be added:
+
+BCP - Best Overall Response of CR/PR by Investigator (confirmation not required)<br>
+CBCP - Best Confirmed Overall Response of CR/PR by Investigator<br>
+A1BOR - Best Overall Response by Investigator (confirmation not required) - RECIST 1.1 adjusted for NED at Baseline<br>
+ACCB - Alternative Confirmed Clinical Benefit by Investigator<br>
+OVRB - Overall Response by BICR<br>
+DEATH - Death<br>
```

Comment:
Feedback from Uwe:
> I would remove this A1BOR, ACCB, DEATH...

Reply:
So not to list PARAMCDs?
The advantage is that with PARAMCD it is easy to find a parameter in Recist 1.1 vignette.
Another option would be to put this at the end:
"iRECIST Best Overall Response of CR/PR by Investigator (confirmation not required) - IBCP"

Reply:
will remove